src/codecov_export.jl

```diff
@@ -0,0 +1,280 @@
+    CoverageUtils.ensure_output_dir(output_file)
+
+    codecov_data = to_codecov_json(fcs)
+
+    open(output_file, "w") do io
+        JSON.print(io, codecov_data)
+    end
+
+    @info "Codecov JSON exported to: $output_file"
+    return abspath(output_file)
+end
+
+"""
+    prepare_for_codecov(fcs::Vector{FileCoverage};
+                        format=:json,
+                        output_dir="coverage",
+                        filename=nothing)
+
+Prepare coverage data for upload with the official Codecov uploader.
+
+# Arguments
+- `fcs::Vector{FileCoverage}`: Coverage data from `process_folder()`
+- `format::Symbol`: Output format (:json or :lcov)
+- `output_dir::String`: Directory to store output files
+- `filename::String`: Custom filename (optional)
+
+# Returns
+- `String`: Path to the generated coverage file
+"""
+function prepare_for_codecov(fcs::Vector{FileCoverage};
+                            format=:json,
+                            output_dir="coverage",
+                            filename=nothing)
+    mkpath(output_dir)
+
+    if format == :json
+        output_file = something(filename, joinpath(output_dir, "coverage.json"))
+        return export_codecov_json(fcs, output_file)
+    elseif format == :lcov
+        # Use existing LCOV functionality
+        output_file = something(filename, joinpath(output_dir, "coverage.info"))
+        LCOV.writefile(output_file, fcs)
+        @info "LCOV file exported to: $output_file"
+        return abspath(output_file)
+    else
+        error("Unsupported format: $format. Supported formats: :json, :lcov")
+    end
+end
+
+"""
+    download_codecov_uploader(; force=false, install_dir=nothing)
+
+Download the official Codecov uploader for the current platform.
+
+# Arguments
+- `force::Bool`: Force re-download even if uploader exists
+- `install_dir::String`: Directory to install uploader (default: temporary directory)
+
+# Returns
+- `String`: Path to the downloaded uploader executable
+"""
+function download_codecov_uploader(; force=false, install_dir=nothing)
+    platform = CoverageUtils.detect_platform()
+    uploader_info = CODECOV_UPLOADERS[platform]
+
+    # Determine installation directory
+    if install_dir === nothing
+        install_dir = mktempdir(; prefix="codecov_uploader_", cleanup=false)
+    else
+        mkpath(install_dir)
+    end
+
+    # Determine executable filename
+    exec_name = platform == :windows ? "codecov.exe" : "codecov"
+    exec_path = joinpath(install_dir, exec_name)
+
+    # Check if uploader already exists
+    if !force && isfile(exec_path)
+        @info "Codecov uploader already exists at: $exec_path"
+        return exec_path
+    end
+
+    @info "Downloading Codecov uploader for $platform..."
+
+    try
+        # Download the uploader
+        Downloads.download(uploader_info.url, exec_path)
+
+        # Make executable on Unix systems
+        if platform != :windows
+            chmod(exec_path, 0o755)
```

Comment:
```suggestion
            chmod(exec_path, 0o555)
```
R-X seems sufficient (I think you should be able to set this on all platforms, and some Windows configurations will require it)